src/modules/escrow/specs/escrow.abi.service.spec.ts

```diff
@@ -0,0 +1,151 @@
+            returnCode: ReturnCode.Ok,
+            returnMessage: '',
+            values: [],
+            firstValue: new U32Value(0),
+        });
+
+        const address = Address.Zero();
+        const permissions = await service.getAddressPermission(
+            address.bech32(),
+        );
+        expect(permissions).toEqual([SCPermissions.NONE]);
+    });
+
+    it('should get admin address permissions', async () => {
+        jest.spyOn(service, 'getGenericData').mockResolvedValue({
+            returnCode: ReturnCode.Ok,
+            returnMessage: '',
+            values: [],
+            firstValue: new U32Value(2),
+        });
+
+        const address = Address.Zero();
+        const permissions = await service.getAddressPermission(
+            address.bech32(),
+        );
+        expect(permissions).toEqual([SCPermissions.ADMIN]);
+    });
+
+    it('should get admin and owner address permissions', async () => {
+        jest.spyOn(service, 'getGenericData').mockResolvedValue({
+            returnCode: ReturnCode.Ok,
+            returnMessage: '',
+            values: [],
+            firstValue: new U32Value(3),
+        });
+
+        const address = Address.Zero();
+        const permissions = await service.getAddressPermission(
+            address.bech32(),
+        );
+        expect(permissions).toEqual([SCPermissions.OWNER, SCPermissions.ADMIN]);
+    });
+
+    it('should get none receivers for sender', async () => {
+        const sender = Address.Zero();
+
+        jest.spyOn(mxGateway, 'getSCStorageKeys').mockResolvedValue({});
+        let receivers = await service.getAllReceivers(sender.bech32());
+        expect(receivers).toEqual([]);
+
+        jest.spyOn(mxGateway, 'getSCStorageKeys').mockResolvedValue({
+            '7065726d697373696f6e73344abc44119cfcace253de05e33c01796c12f96f3bcc52b504b9bc2b96927ceb':
+                '03',
+            '7065726d697373696f6e7360bb6011a781eb8b53d61d79047450adaa46977e20239312951262c1057d1090':
+                '02',
+            '73656e6465724c6173745472616e7366657245706f6368344abc44119cfcace253de05e33c01796c12f96f3bcc52b504b9bc2b96927ceb':
+                '1143',
+            '73656e6465724c6173745472616e7366657245706f6368ba66e8a916236723bf1f94996488b6fc57dc5f6ec0512b9b74b53ef65f9508cb':
+                '1139',
+        });
+
+        receivers = await service.getAllReceivers(sender.bech32());
+        expect(receivers).toEqual([]);
+
+        jest.spyOn(mxGateway, 'getSCStorageKeys').mockResolvedValue({
+            '616c6c53656e646572732c5594ae2f77a913119bc9db52833245a5879674cd4aeaedcd92f6f9e7edf17d2e696e646578344abc44119cfcace253de05e33c01796c12f96f3bcc52b504b9bc2b96927ceb':
+                '01',
+            '616c6c53656e646572732c5594ae2f77a913119bc9db52833245a5879674cd4aeaedcd92f6f9e7edf17d2e6974656d00000001':
+                '6e593caf5c21cd2e419c8249101e20eb53770d6b2512ee9f19b971fd3c0a0e89', // erd1devnet6uy8xjusvusfy3q83qadfhwrtty5fwa8ceh9cl60q2p6ysra7aaa
```

Comment:
maybe remove comment

Reply:
I added it to know the encoded address I put there